src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
 export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
 
+  getArea(): number;
 }
 
 export class Triangle {
+  shape = 'triangle';
 
+  constructor(
+    public color: string,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (sideA <= 0
+      || sideB <= 0
+      || sideC <= 0
+    ) {
+      throw new Error('side must be greater than zero');
+    }
```

Comment:
Better create a helper function that will accept arguments (any length) and determine if some have `value <= 0`.
```
(...args) => args.some() ....
```
And you could use it for any figure you need to check. You have almost the same check in 3 places all over the code.